### PR TITLE
Type alias handles

### LINKED ISSUE #1415: [Handles] Use type aliases for different Handle types
- [x] just to be type clear on display vs base vs canonical and make sure they are in the right locations

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is  to define type aliases for different handle types for clarity

Closes #1415 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -429,7 +429,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: spec_name!("frequency"),
 	impl_name: create_runtime_str!("frequency"),
 	authoring_version: 1,
-	spec_version: 28,
+	spec_version: 29,
```
Comment (wilwade) [inline runtime/frequency/src/lib.rs:432]: I wouldn't have expected this to be needed.

File: common/primitives/src/handles.rs [hunk 2]
```diff
@@ -22,9 +22,14 @@ pub const SUFFIX_MAX_DIGITS: usize = 5; // The max value of a HandleSuffix (u16)
...
-pub type Handle = BoundedVec<u8, ConstU32<HANDLE_BASE_BYTES_MAX>>;
-
+/// A generic handle type  (base, canonical, or display)
+type Handle = BoundedVec<u8, ConstU32<HANDLE_BASE_BYTES_MAX>>;
+/// A base handle, which is chosen by the user
+pub type BaseHandle = Handle;
+/// A canonical base, which is a reduced/translated version of the base handle
+pub type CanonicalBase = Handle;
+/// A display handle, which is a base handle with a suffix
+pub type DisplayHandle = Handle;
```
Comment (saraswatpuneet) [inline common/primitives/src/handles.rs:32]: @wilwade  you mean this?

**Review by wilwade — APPROVED** [review]
One question on the spec_version update, but all the rest is great. So much nicer to talk in types.

File: common/primitives/src/handles.rs (hunk 2 above)
Comment (wilwade) [reply common/primitives/src/handles.rs:32]: Yes. This is great.

File: runtime/frequency/src/lib.rs (hunk 1 above)
Comment (saraswatpuneet) [reply runtime/frequency/src/lib.rs:432]: The CI broke since it is going in main

Comment (saraswatpuneet) [reply runtime/frequency/src/lib.rs:432]: https://github.com/LibertyDSNP/frequency/actions/runs/4733631145

File: common/primitives/src/handles.rs [hunk 3]
```diff
@@ -22,9 +22,15 @@ pub const SUFFIX_MAX_DIGITS: usize = 5; // The max value of a HandleSuffix (u16)
...
-
+/// A generic handle type  (base, canonical, or display)
+type Handle = BoundedVec<u8, ConstU32<HANDLE_BASE_BYTES_MAX>>;
+/// A base handle, which is chosen by the user
+pub type BaseHandle = Handle;
+/// A canonical base, which is a reduced/translated version of the base handle
+pub type CanonicalBase = Handle;
+/// A display handle, which is a base handle with suffix separated by a delimiter
+pub type DisplayHandle =
+	BoundedVec<u8, ConstU32<{ HANDLE_BASE_BYTES_MAX + SUFFIX_MAX_DIGITS as u32 + 1u32 }>>;
```
Comment (saraswatpuneet) [inline common/primitives/src/handles.rs:33]: @wilwade  what you think

File: pallets/handles/src/tests/handle_creation_tests.rs [hunk 4]
```diff
@@ -293,3 +293,33 @@ fn claim_handle_fails_when_handle_contains_unsupported_unicode_characters() {
...
+		assert_eq!(handle_result.base_handle, "𝔸".repeat(8).as_bytes().to_vec());
+		assert!(handle_result.suffix > 0);
+		let display_handle = "𝔸".repeat(8) + "." + &handle_result.suffix.to_string();
+		let display_handle_vec = display_handle.as_bytes().to_vec();
+		let msa_id_from_state =
+			Handles::get_msa_id_for_handle(display_handle_vec.try_into().unwrap());
+		assert!(msa_id_from_state.is_some());
+		assert_eq!(msa_id_from_state.unwrap(), msa_id);
+	});
+}
```
Comment (saraswatpuneet) [inline pallets/handles/src/tests/handle_creation_tests.rs:325]: This test claims all 32b of allowed handle space and check the integrity of state especially checking if DisplayHandle is correctly formed

**Review by johnmela — APPROVED** [review]

**Pull request merged**